arithmetization/src/main/java/net/consensys/linea/blockcapture/snapshots/StorageSnapshot.java

```diff
@@ -19,17 +19,36 @@
 
 import org.apache.tuweni.units.bigints.UInt256;
 import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.evm.account.Account;
 import org.hyperledger.besu.evm.worldstate.WorldView;
 
 public record StorageSnapshot(String address, String key, String value) {
   public static Optional<StorageSnapshot> from(
       Address address, UInt256 key, final WorldView world) {
-    return Optional.ofNullable(world.get(address))
-        .map(
-            account ->
-                new StorageSnapshot(
-                    address.toHexString(),
-                    key.toHexString(),
-                    account.getStorageValue(key).toHexString()));
+    // Lookup account that was touched
+    Account account = world.get(address);
+    // Check account *really* exists from a storage perspective.  This is important to distinguish
+    // for accounts which are created *during* the conflation.  Such accounts may have technically
+    // existed before the conflation (e.g. they had a non-zero balance) but could still have been
+    // "created" during the conflation.  In such case, this snapshot would be simply assigning 0x0
+    // to the given storage locations.  However, we don't want to create a storage snapshot in such
+    // cases, as this then leads to a CREATE[2] failure when executing the conflation.
+    if (accountExists(account)) {
+      // Accounts exists, so create snapshot.
+      return Optional.of(
+          new StorageSnapshot(
+              address.toHexString(),
+              key.toHexString(),
+              account.getStorageValue(key).toHexString()));
+    } else {
+      return Optional.empty();
+    }
+  }
+
+  private static boolean accountExists(final Account account) {
```

Comment:
AFAIU the EYP defines for an account $a$
- `existence` as existence in the state, i.e. $\sigma[a] \neq \varnothing$
- `EMPTY` as having nontrivial nonce, balance or code
- `DEAD` as either not existing or being empty
- prohibited from being deployed at as existing and either having nontrivial nonce, code or storage


<img width="699" alt="image" src="https://github.com/user-attachments/assets/a40007cc-2253-4148-b172-f4d5c5bc873c">

<img width="673" alt="image" src="https://github.com/user-attachments/assets/b722e975-87db-4d72-998d-a5ab87d1dfa5">

along with [EIP-7610](https://eips.ethereum.org/EIPS/eip-7610)

So maybe a more accurate name would be `accountMayBeDeployedAt` or so.